### ISSUE TITLE
Attach to Golang functions

### ISSUE THREAD

**Issue by jasonkeene** [iovisor/bpftrace#168]
All go functions are compiled to symbols that have a period in them to delimit what package they are defined in. For example:

```
$ objdump -t bpfgotest  | grep main.add
0000000000451400 g     F .text  0000000000000019 main.add
```

I would like to be able to attach to these with uprobe/uretprobe however I get the following error:

```
$ bpftrace -e 'uprobe:/root/bpfgotest:main.add { printf("a\n"); }'
1.28: syntax error, unexpected ., expecting {
$ bpftrace -e 'uretprobe:/root/bpfgotest:main.add { printf("a\n"); }'
1.31: syntax error, unexpected ., expecting {
```

I tried escaping the period and using a wildcard but that doesn't work with uprobe/uretprobe:

```
$ bpftrace -e 'uprobe:/root/bpfgotest:main\.add { printf("a\n"); }'
1.28: invalid character '\'
1.28-29: syntax error, unexpected ., expecting {
$ bpftrace -e 'uprobe:/root/bpfgotest:*add { printf("a\n"); }'
Wildcard matches aren't available on probe type 'uprobe'
No probes to attach
```

Do you know how I might go about attaching to Golang funcs? Is this possible currently?

Also, I imagine I will have to do some trickery to get the args/ret values since the ABI for AMD64 go is not the SystemV one. I don't mind doing some work to get that working though.

**Comment by brendangregg** [comment]
also see #166

**Comment by jasonkeene** [comment]
Closing this as it was solved by #166 I will open another issue about supporting Go's calling convention.
